skbio/alignment/_tabular_msa.py

```diff
@@ -308,44 +318,23 @@ def from_dict(cls, dictionary):
         return cls(viewvalues(dictionary), keys=viewkeys(dictionary))
 
     @experimental(as_of='0.4.0-dev')
-    def __init__(self, sequences, metadata=None, key=None, keys=None):
+    def __init__(self, sequences, metadata=None, minter=None, keys=None):
         sequences = iter(sequences)
-        seq = next(sequences, None)
-
-        dtype = None
-        length = 0
-        seqs = []
-        if seq is not None:
-            seqs.append(seq)
-            dtype = type(seq)
-            if not issubclass(dtype, IUPACSequence):
-                raise TypeError(
-                    "`sequences` must contain scikit-bio sequence objects "
-                    "that have an alphabet, not type %r" % dtype.__name__)
-            length = len(seq)
-
-            for seq in sequences:
-                if type(seq) is not dtype:
-                    raise TypeError(
-                        "`sequences` cannot contain mixed types. Type %r does "
-                        "not match type %r" %
-                        (type(seq).__name__, dtype.__name__))
-                if len(seq) != length:
-                    raise ValueError(
-                        "`sequences` must contain sequences of the same "
-                        "length: %r != %r" % (len(seq), length))
-                seqs.append(seq)
-
-        self._seqs = seqs
-        self._dtype = dtype
-        self._shape = _Shape(sequence=len(seqs), position=length)
+
+        self._seqs = []
+        self._dtype = None
+        self._shape = _Shape(sequence=0, position=0)
+        self._minter = None
+
+        for seq in sequences:
```

Comment:
Good catch, `iter` was previously necessary because we were calling `next` but now we're only looping. This can be removed.